gitindex/delete.go

```diff
@@ -19,7 +20,9 @@ func DeleteRepos(baseDir string, urlPrefix *url.URL, names map[string]struct{},
 	var toDelete []string
 	for _, p := range paths {
 		_, exists := names[p]
-		if filter.Include(filepath.Base(p)) && !exists {
+		repoName := strings.Replace(p, filepath.Join(urlPrefix.Host, urlPrefix.Path), "", 1)
+		repoName = strings.TrimPrefix(repoName, "/")
+		if filter.Include(repoName) && !exists {
```

Comment:
Can you give an example where `filepath.Base` fails? Please add your example to the existing test in delete_test.go, too.